src/control/conf.c

```diff
@@ -574,8 +671,40 @@ gboolean dt_confgen_get_bool(const char *name, dt_confgen_value_kind_t kind)
 
 float dt_confgen_get_float(const char *name, dt_confgen_value_kind_t kind)
 {
+  if(!dt_confgen_value_exists(name, kind))
+  {
+    //early bail
+    switch (kind)
+    {
+    case DT_MIN:
+      return -FLT_MAX;
+      break;
+    case DT_MAX:
+      return FLT_MAX;
+      break;
+    default:
+      return 0.0f;
+      break;
+    }
+  }
   const char *str = dt_confgen_get(name, kind);
```

Comment:
At this stage we know it does not exist... Why retrieve str and call solve?

all returns:
`return isnan(value) ? -FLT_MAX : (value > 0 ? value + 0.5f : value - 0.5f);`
can just be:
`return -FLT_MAX;`

Just put an else here! Or am I missing something ?

Reply:
Sorry!!!! I hadn't see the !  :)

Reply:
So the code is perfect... Read too fast!

Reply:
Noooo! it's not perfect! float is wrong!!!

plz change to:

```c
float dt_confgen_get_float(const char *name, dt_confgen_value_kind_t kind)
{
  if(!dt_confgen_value_exists(name, kind))
  {
    //early bail
    switch (kind)
    {
    case DT_MIN:
      return -FLT_MAX;
      break;
    case DT_MAX:
      return FLT_MAX;
      break;
    default:
      break;
    }
    return 0.0f;
  }
  
  const char *str = dt_confgen_get(name, kind);

  //if str is NULL or empty, dt_calculator_solve will return NAN
  const float value = dt_calculator_solve(1, str);

  switch (kind)
  {
  case DT_MIN:
    // to anyone askig FLT_MIN is superclose to 0, not furthest value from 0 possible in float
    return isnan(value) ? -FLT_MAX : value;
    break;
  case DT_MAX:
    return isnan(value) ? FLT_MAX : value;
    break;
  default:
    break;
  }
  return isnan(value) ? 0.0f : value;
}
```

Reply:
Indeed, missed that...